### PR TITLE
Add dev k8s manifest

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context
We want to start testing the new service in K8s so a dev manifest can live in this repo for now (when it replaces the python service this will live in the kubernetes repo instead).

# What has changed
* Add dev k8s manifest
* Fix docker image name to use the rm directory as other services do
* Use more consistent env var names

# How to test?
Apply this deployment yml in a dev environment, run an acceptance test against it.

# Links
https://trello.com/c/Z9KbRyyY/898-pubsub-adapter-get-it-running-in-k8s-3

### CODE REVIEW

**Comment by AdamHawtin** [comment]
[Pubsub Adapter - Get it running in k8s (3?)](https://trello.com/c/Z9KbRyyY/898-pubsub-adapter-get-it-running-in-k8s-3)

File: pubsub-adapter-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            - name: QM_UNDELIVERED_SUBSCRIPTION_NAME
+              value: "rm-qm-undelivered-subscription"
+            - name: QM_UNDELIVERED_SUBSCRIPTION_PROJECT
+              valueFrom:
+                configMapKeyRef:
+                  name: project-config
+                  key: project-name
+            - name: GOOGLE_APPLICATION_CREDENTIALS
+              value: "/gcp-credentials/service-account-key.json"
+            - name: LOG_LEVEL
```
Comment (davidmort) [inline pubsub-adapter-deployment.yml]: is LOG_LEVEL used at the moment?

**Review by davidmort — CHANGES_REQUESTED** [review]

File: pubsub-adapter-deployment.yml (hunk 1 above)
Comment (AdamHawtin) [reply pubsub-adapter-deployment.yml]: nope, removing

File: pubsub-adapter-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  selector:
+    matchLabels:
+      app: pubsub-adapter
+  template:
+    metadata:
+      name: pubsub-adapter
+      labels:
+        app: pubsub-adapter
+        env: dev
+        version: "0.1.0dev"
```
Comment (NickGrantX) [inline pubsub-adapter-deployment.yml:24]: we usually have v1.0 here don't we?

**Review by NickGrantX — APPROVED** [review]
LGTM 👍

File: pubsub-adapter-deployment.yml (hunk 2 above)
Comment (AdamHawtin) [reply pubsub-adapter-deployment.yml:24]: Yeah, we're not really making proper use of this label field anywhere, I just thought I'd make sure it was different for this "dev" version of the manifest.

**Review by davidmort — APPROVED** [review]

**Review by hbrace — APPROVED** [review]
Works for me 👍

**Pull request merged**